net/httpclient_test.go

```diff
@@ -245,13 +246,29 @@ func TestTransport(t *testing.T) {
 			req:         httptest.NewRequest("GET", "http://example.com/", nil),
 			wantErr:     false,
 		},
+		{
+			name:       "context deadline error should occur when the request is timedout",
+			req:        httptest.NewRequest("GET", "http://example.com/", nil),
+			wantErr:    true,
+			reqTimeout: 10 * time.Millisecond,
+		},
+		{
+			name:       "no context deadline error should occur when the request is processed within the requested timeout",
+			req:        httptest.NewRequest("GET", "http://example.com/", nil),
+			wantErr:    false,
+			reqTimeout: 5 * time.Millisecond,
+		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
 			s := startTestServer(func(r *http.Request) {
 				if r.Method != tt.req.Method {
 					t.Errorf("wrong request method got: %s, want: %s", r.Method, tt.req.Method)
 				}
 
+				if tt.reqTimeout != 0 && tt.wantErr {
+					time.Sleep(tt.reqTimeout + 10*time.Millisecond)
+				}
```

Comment:
Lets have two independent values: tt.reqTimeout for client and tt.serverDelay.
Then we can have two test cases:
* reqTimeout > serverDelay
* reqTimeout < serverDelay